src/lib.rs

```diff
@@ -0,0 +1,124 @@
+pub enum VersionLabel {
+  Patch,
+  Minor,
+  Major,
+}
+
+pub fn update_version_by_label(
+  cargo_toml_content: String,
+  label: VersionLabel,
+) -> String {
+  let current_version = get_version(&cargo_toml_content);
+  let major: i32 = current_version.0.parse().unwrap();
+  let minor: i32 = current_version.1.parse().unwrap();
+  let patch: i32 = current_version.2.parse().unwrap();
+
+  let new_version = match label {
+    VersionLabel::Major => format!("{}.0.0", major + 1),
+    VersionLabel::Minor => format!("{}.{}.0", major, minor + 1),
+    VersionLabel::Patch => format!("{}.{}.{}", major, minor, patch + 1),
+  };
+
+  cargo_toml_content.replace(&tuple_to_string(current_version), &new_version)
+}
+
+pub fn update_version(cargo_toml_content: String, version: String) -> String {
+  let current_version_tuple = get_version(&cargo_toml_content);
+  let current_version = tuple_to_string(current_version_tuple);
+  cargo_toml_content.replace(&current_version, &version)
+}
+
+fn tuple_to_string(tuple_version: (String, String, String)) -> String {
+  tuple_version.0 + "." + &tuple_version.1 + "." + &tuple_version.2
+}
+
+fn get_version(cargo_toml_content: &str) -> (String, String, String) {
+  let version = cargo_toml_content
+    .lines()
+    .filter(|line| line.contains("version"))
+    .map(|line| {
+      line
+        .replace("version", "")
+        .replace('=', "")
+        .replace('"', "")
+        .trim()
+        .to_string()
+    })
+    .next();
+
+  let version = match version {
+    Some(v) => v,
+    None => panic!("Your Cargo.toml file does not have a \"version\" entry"),
+  };
```

Comment:
nit: this `match` can be replaced with `expect` 

```suggestion
  let version = version.expect("Your Cargo.toml file does not have a \"version\" entry");
```

Reply:
Nice! I just used `match` and `panic` here for now, but the main idea is to return a `Result` and at `main.rs` file use `unwrap_or_else` to show a more friendly error message, instead of `panic!` xD

But good tip. Thanks!